### ISSUE TITLE
[Feature] Ability to use d3 time intervals instead of 'numberOfTicks' on xAxis (d3.timeDay.every(step))

### ISSUE THREAD

**Issue by ajacquierbret** [JesperLekland/react-native-svg-charts#484]
### Ability to use d3 time intervals instead of 'numberOfTicks' on xAxis (d3.timeDay.every(step)) :

It could be really helpful to use time intervals instead of d3 native ticks calculation, especially with xAxis time series.
I'm facing "an issue" with this prop. Ticks don't behave as I would concerning dates based axis, but where time intervals could be the best way to handle time series.

I need to provide consistent time intervals between every xAxis value and the 'numberOfTicks' prop doesn't provide a way to do it, while 'd3.timeDay.every' could be a really helpful workaround.

Actually, I need to display a date value "every four hours", but if I set the 'numberOfTicks' to '6' (6x4 = 24h) I get eight points on the xAxis :

<img width="375" alt="Capture d’écran 2020-07-02 à 09 58 26" src="https://user-images.githubusercontent.com/45269819/86332714-2fe00100-bc4b-11ea-928c-0bfd7ba4bc65.png">

Here's a link to a post explaining this behaviour : http://www.d3noob.org/2016/08/changing-number-of-ticks-on-axis-in.html

By the way, huge thanks to @JesperLekland  for this amazing work  ! 🥇

**Comment by ajacquierbret** [comment]
Ok I found a workaround that doesn't involve time intervals, I just checked every label index and used a modulus (%) operator to "do nothing" with an index that wouldn't be equal to zero on a `index / x = 0` division. But anayway, this feature request is still open because I think it could be great to have the ability to use d3 time intervals.

```javascript
// Chart.tsx
formatXLabel={(label, index) =>
            `${getFormatedLabel(label, index)}`
          }
```

```javascript
// Utils.tsx
export function getFormatedLabel(label, index) {
  if (index % 2) {
    return moment(label).format('HH [h]');
  } else {
    return '';
  }
}
```

<img width="363" alt="Capture d’écran 2020-07-02 à 10 21 54" src="https://user-images.githubusercontent.com/45269819/86334767-ed6bf380-bc4d-11ea-8e6a-a4026bf553c2.png">

**Comment by JesperLekland** [comment]
I agree that `d3.timeDay.every` looks like a potential valuable addition. I don't maintain this library actively anymore though so I would love a PR with said feature. I'm glad you found a workaround in the meantime though, and thank you for sharing that with the community.

**Comment by ajacquierbret** [comment]
> I agree that `d3.timeDay.every` looks like a potential valuable addition. I don't maintain this library actively anymore though so I would love a PR with said feature. I'm glad you found a workaround in the meantime though, and thank you for sharing that with the community.

I'll try to do some PR if I can find the time to dig into this library, thanks a lot for your very fast answer ! You're welcome !

**Comment by kchangch** [comment]
Hi man, it's me again. I'm trying to make a barchart now and I'm having issues aligning the grid with the Y-Axis. Is there any tips you could give me?

<img width="368" alt="Screen Shot 2020-09-01 at 12 02 30 AM" src="https://user-images.githubusercontent.com/43304067/91812712-7075df80-ebe6-11ea-8991-0ce8ad3b0b0c.png">

I have tried already all the ways to adjust the contentInset of the axis to be the same as the barchart but no luck.

**Comment by gigby** [comment]
@ajacquierbret
Actually, you can simulate d3.every behavior. formatXLabel is a part of the solution. Actually you just need return "pseudoIndex" in "xAccessor" prop of <LineChart />.

The code should look like this
`const firstTime = data[0].time; // first time of displaying events array
  const lastTime = data[data.length - 1].time;
  const tDiffFromN = (lastTime - firstTime) / (data.length - 1); // time for one tick
  
// and finally
xAccessor={({ item, index }) => {
            console.tron.log('XXX', item);
            // return index; // DO NOT DO IT for this case
            const pseudoIndex = (item.time - firstTime) / tDiffFromN; // construct pseudo index and be happy
            return pseudoIndex;
          }}
  
`
Yellow lines are ticks. It's the right way to process random time on X axis.
![Screenshot 2022-10-12 at 20 27 49](https://user-images.githubusercontent.com/18383850/195397844-e678a6fb-39ef-40c2-94ab-1b07526fd782.png)

This is the wrong way to use the library for time-showing. Look at the curve :-)

![Screenshot 2022-10-12 at 20 30 31](https://user-images.githubusercontent.com/18383850/195398492-ab3f1772-62cc-4a9d-a192-c0357dd53cfe.png)

PS: I seem this thread is outdated but I saw a few "bug reports" from people who are stuck on this
